UIKit/ML App UIKit/ViewController.swift

```diff
@@ -7,13 +7,220 @@
+        
     }
+    
+    private func reloadData(product: Product) {
+        DispatchQueue.main.async {
+            self.product = product
+            self.tableView.reloadData()
+        }
+    }
+    
+    func showAlert() {
+        let alert = UIAlertController(title: "Shit Happened", message: "Some Shit Happened", preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Well...Shit", style: .default, handler: nil))
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+}
 
+// MARK: - UITableViewDelegate
+extension ViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+       // print(products[indexPath.row])
+    }
+}
 
+// MARK: - UITableViewDataSource
+extension ViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return product.results.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = UITableViewCell()
+        cell.textLabel?.text = product.results[indexPath.row].title
+//        cell.textLabel?.text = product.results[indexPath.row].condition
+//        cell.textLabel?.text = String(product.results[indexPath.row].price)
+//        cell.textLabel?.text = product.results[indexPath.row].permalink
+//        cell.textLabel?.text = product.results[indexPath.row].thumbnail
+        return cell
+    }
 }
 
+
+//  MARK: - Get data from searchbar
+
+
+
+extension ViewController: UISearchBarDelegate {
+    
+    public func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        dispatchWorkItem?.cancel()
+        let workItem = DispatchWorkItem { [weak self] in
+            
+            print("\(searchText)")
+            self?.networkManager.fetchData(for: "https://api.mercadolibre.com/sites/MLM/search?q=\(searchText)") { result in
+                switch result {
+                case .success(let data): self?.reloadData(product: data)
+                case .failure(let error):
+                    print(error)
+                    print("error")
+                    
+                    DispatchQueue.main.async {
+                        self?.showAlert()
+                    }
+                }
+            }
+        }
+        
+        dispatchWorkItem = workItem
+        if let waitTime = dispatchWorkItem {
+            queue.asyncAfter(deadline: .now() + 2.5, execute: waitTime)
+        }
+
+    }
+}
+
+// MARK: - Alert
+
+
+
+
+// MARK: - Hide Keyboard
+
+extension UIViewController {
+    func hideKeyboardWhenTappedAround() {
+        let tap: UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(UIViewController.dismissKeyboard))
+        tap.cancelsTouchesInView = false
+        view.addGestureRecognizer(tap)
+    }
+
+    @objc func dismissKeyboard() {
+        view.endEditing(true)
```

Comment:
this should be on the class definition, it should not be on a separate extension ..also private functions